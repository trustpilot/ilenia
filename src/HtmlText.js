import PropTypes from 'prop-types';
import Parser from 'html-react-parser';
import withTranslations from './withTranslations';
import interpolate from './interpolate';

const HtmlText = ({ textKey, interpolations = {}, translations = {}, tag }) => {
  let string = translations[textKey];
  if (!string) {
    console.error(`Couldn't find '${textKey}' in the translations table`); // eslint-disable-line no-console
    return null;
  }

  string = interpolate(string, interpolations, tag);
  return Parser(string);
};

HtmlText.propTypes = {
  textKey: PropTypes.string.isRequired,
  interpolations: PropTypes.objectOf(PropTypes.string),
  tag: PropTypes.shape({
    start: PropTypes.string,
    end: PropTypes.string,
  }),
};

export default withTranslations(HtmlText);
