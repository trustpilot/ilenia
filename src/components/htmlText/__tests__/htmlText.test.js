import React from 'react';
import renderer from 'react-test-renderer';
import LocalizationProvider from '../../provider';
import HtmlText from '../';

const translations = {
  'test1': 'Just a random string',
  'test2': 'A string but with {html} in it',
  'test3': 'A string that is sort of {html1}italic{html2} and {html3}bold{html4} and such',
  'test4': 'A string that is sort of {html1}italic{html2}',
  'test5': 'A string that uses [smtg1]another[smtg2] placeholder',
  'test6': 'A string that has <b>regular HTML</b> in it',
};

const setupText = (string, interpolations, tag) => (
  <LocalizationProvider locale="en-US" translations={translations}>
    <HtmlText id={string} interpolations={interpolations} tag={tag}/>
  </LocalizationProvider>
);

test('Renders a plain string', () => {
  const component = renderer.create(setupText('test1'));

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Renders a string with one HTML part', () => {
  const component = renderer.create(setupText('test2', {html: '<span>hello</span>'} ));

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Renders a string with multiple HTML parts in it', () => {
  const component = renderer.create(setupText('test3', {
    html1: '<em>',
    html2: '</em>',
    html3: '<b>',
    html4: '</b>',
  }));

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Renders a string with HTML parts in the end', () => {
  const component = renderer.create(setupText('test4', {
    html1: '<em>',
    html2: '</em>',
  }));

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Renders a string with HTML marked with another key', () => {
  const component = renderer.create(setupText('test5', {
    smtg1: '<b>',
    smtg2: '</b>',
  }, {
    start: '[',
    end: ']',
  }));

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});


test('Renders a string with HTML in it', () => {
  const component = renderer.create(setupText('test6'));

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Renders nothing if the string is not there', () => {
  console.error = jest.fn(); // eslint-disable-line no-console

  const component = renderer.create(setupText('test7'));

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  expect(console.error).toHaveBeenCalled(); // eslint-disable-line no-console
});
