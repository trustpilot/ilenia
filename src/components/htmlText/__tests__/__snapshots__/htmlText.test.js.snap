// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders a plain string 1`] = `
<span
  dangerouslySetInnerHTML={
    Object {
      "__html": "Just a random string",
    }
  }
/>
`;

exports[`Renders a string with HTML in it 1`] = `
<span
  dangerouslySetInnerHTML={
    Object {
      "__html": "A string that has <b>regular HTML</b> in it",
    }
  }
/>
`;

exports[`Renders a string with HTML marked with another key 1`] = `
<span
  dangerouslySetInnerHTML={
    Object {
      "__html": "A string that uses <b>another</b> placeholder",
    }
  }
/>
`;

exports[`Renders a string with HTML parts in the end 1`] = `
<span
  dangerouslySetInnerHTML={
    Object {
      "__html": "A string that is sort of <em>italic</em>",
    }
  }
/>
`;

exports[`Renders a string with multiple HTML parts in it 1`] = `
<span
  dangerouslySetInnerHTML={
    Object {
      "__html": "A string that is sort of <em>italic</em> and <b>bold</b> and such",
    }
  }
/>
`;

exports[`Renders a string with one HTML part 1`] = `
<span
  dangerouslySetInnerHTML={
    Object {
      "__html": "A string but with <span>hello</span> in it",
    }
  }
/>
`;

exports[`Renders nothing if the string is not there 1`] = `null`;
